File: 05_FizzBuzz/fizzbuzz.rb
# acskii :p

def fizzbuzz(range)
  if range.is_a? Range 
    for i in range
      print "#{i}"
      if (i % 3 == 0) && (i % 5 == 0)
        print " - FizzBuzz\n"
      elsif i % 3 == 0
        print " - Fizz\n"
      elsif i % 5 == 0
        print " - Buzz\n"
      else 
        print "\n"
      end
    end
  else
    puts "Only a range is accepted as an argument."
  end
end

fizzbuzz(1..15)
puts "-"*10
fizzbuzz(1..25)
puts "-"*10
fizzbuzz(10..15)
puts "-"*10

File: 05_FizzBuzz/fizzbuzz.js
// acskii :p

function fizzbuzz(sample) {
    // sample is an array of numbers
    if (Array.isArray(sample)) {
        for (const i of sample) {
            if (Number.isInteger(i)) {
                if (i % 15 == 0) {
                    console.log(`${i} - FizzBuzz`);
                } else if (i % 3 == 0) {
                    console.log(`${i} - Fizz`);
                } else if (i % 5 == 0) {
                    console.log(`${i} - Buzz`);
                } else {
                    console.log(`${i}`); 
                }
            } else {
                console.log("null\n");
            }
        }
    } else {
        console.log("Only an array is accepted as an argument");
        return null;
    }
}